docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,250 @@ sidebar_position: 7
 
-```bash   showLineNumbers
-git pull upstream master
-```
+Most often, code reviews are just a matter of the maintainer enforcing consistency in the codebase, or spotting bugs you might have missed (like we all do), or asking to make code clearer for someone that has never read it before. Always feel free to ask clarifying questions, or to push back if you disagree with something.
 
-You can now go back to step 1 and make new changes all over again.
+Once your pull request is merged (aka accepted in the main codebase), you can delete your branch. Github has a easy button for you to do so.
 
-## Extra: Use a different GitHub account
+#### How to Make Requested Changes
 
-If you already have a GitHub account set up on your machine and want to use a different one for BobaBoard, you can use the following instructions:
+When changes are requested, you will likely see comments on the pull request you have submitted. Simply go back to step 2, write code, commit and push until all comments are fixed. Then go back to the pull request, check that everything is in order, and ask for an additional review. You should mark the comments you have addressed as done by using the "resolve convo" button.
 
-### 0. Create a special folder to house BobaBoard code
+### 7. Pulling from upstream
 
-This is the folder you're going to `clone` the BobaBoard repositories in.
+Now that the feature code has been merged to the `upstream master` branch, the code in the `master` branch of the developer's fork (which doesn't contain the original feature) is out of date. To download the updated code, the developer "pulls" the changes from the `upstream` codebase. This copies all the new code commits in the `upstream` codebase to the developer's.
 
-### 1. Set Up your Git username and email
+...
 
-In your new folder, run the following commands to set up the new user:
+When you go back to the codebase, it's likely that more changes will have been made on the original BobaBoard codebase your code is forked from. To update your git fork to reflect the changes of "upstream" (aka the original codebase) run the following command.
 
 ```bash   showLineNumbers
-git config user.name YOUR_BOBABOARD_GITHUB_USERNAME_HERE
-git config user.email YOUR_BOBABOARD_GITHUB_EMAIL_HERE
+git pull upstream master
 ```
 
-:::warning
-The email you set up will be visible to the world. GitHub provides a secret email address in your account section. Use that!
-:::
+You can now go back to step 1 and make new changes all over again.
 
-### 2. Update the `origin` of the repository to the right username
+### 8. Working on a new feature
 
-```
-git remote set-url origin git@github.com-YOUR_BOBABOARD_GITHUB_USERNAME_HERE:YOUR_REMOTE_URL_HERE
-```
+To work on a new feature, the developer repeats the same process starting from step 1.
```

Comment:
**My Final Thoughts**
Not really related to the sentence above, but more a style observation. We don't really use

1. Title
    > blockquote

Anywhere else in the docs, and semantically it might be confusing if people use screenreaders. You should be able to create properly indented subparagraphs by doing:

```markdown
1. Title

  Description paragraphs.
  ```
_Note: I can't remember right now if it's two spaces or four, but I think it's two. I do know the blank line between the number and the paragraph is key._

I'm also not certain in the Guide about making the steps a header. It does allow people to jump to specific parts, but it's not something we do anywhere else in the docs. If you're just trying to make them stand out more I'd use bold text for them.

Overall, this is a really solid bit of documentation!

Reply:
The format is based on the previous version so I kinda wrote on top, it worked for me but we could scrap it altogether if it's not helpful or just do `**Title**: text from the blockquote` and make some chonky list items.

In regards to the steps being headers, do we.. actually have guides/tutorials in the docs already? I assumed we didn't so I didn't bother looking, oops. We could also split the overview and the guide proper into different pages? Some reasearch is probably required steal some ideas. 

We should ask some of the people we onboarded this sprint to comment too 👀